improver_tests/calibration/reliability_calibration/test_ManipulateReliabilityTable.py

```diff
@@ -199,6 +448,66 @@ def test_no_change(self):
         assert_array_equal(result[1].data, self.multi_threshold_rt[1].data)
         self.assertEqual(result[1].coords(), self.multi_threshold_rt[1].coords())
 
+    def test_combine_undersampled_bins_monotonic(self):
+        """Test expected values are returned when a bin is below the minimum
+        forecast count when the observed frequency is monotonic."""
+
+        expected_data = np.array(
+            [[0, 250, 425, 1000], [0, 250, 425, 1000], [1000, 1000, 600, 1000]]
+        )
+        expected_bin_coord_points = np.array([0.1, 0.3, 0.6, 0.9], dtype=np.float32)
+        expected_bin_coord_bounds = np.array(
+            [[0.0, 0.2], [0.2, 0.4], [0.4, 0.8], [0.8, 1.0]], dtype=np.float32,
+        )
+        self.multi_threshold_rt.data[1] = np.array(
+            [
+                [0, 250, 50, 375, 1000],  # Observation count
+                [0, 250, 50, 375, 1000],  # Sum of forecast probability
+                [1000, 1000, 100, 500, 1000],  # Forecast count
+            ]
+        )
+
+        result = Plugin().process(self.multi_threshold_rt.copy())
+        assert_array_equal(result[0].data, self.multi_threshold_rt[0].data)
+        self.assertEqual(result[0].coords(), self.multi_threshold_rt[0].coords())
+        assert_array_equal(result[1].data, expected_data)
+        assert_allclose(
+            result[1].coord("probability_bin").points, expected_bin_coord_points
+        )
+        assert_allclose(
+            result[1].coord("probability_bin").bounds, expected_bin_coord_bounds
+        )
+
+    def test_combine_undersampled_bins_non_monotonic(self):
+        """Test expected values are returned when a bin is below the minimum
+        forecast count when the observed frequency is non-monotonic."""
+
+        expected_data = np.array(
+            [[0, 250, 425, 1000], [0, 250, 425, 1000], [1000, 1000, 600, 1000]]
+        )
+        expected_bin_coord_points = np.array([0.1, 0.3, 0.6, 0.9], dtype=np.float32)
+        expected_bin_coord_bounds = np.array(
+            [[0.0, 0.2], [0.2, 0.4], [0.4, 0.8], [0.8, 1.0]], dtype=np.float32,
+        )
+        self.multi_threshold_rt.data[1] = np.array(
+            [
+                [0, 250, 50, 375, 1000],  # Observation count
+                [0, 250, 50, 375, 1000],  # Sum of forecast probability
+                [1000, 1000, 100, 500, 1000],  # Forecast count
+            ]
+        )
+
+        result = Plugin().process(self.multi_threshold_rt.copy())
+        assert_array_equal(result[0].data, self.multi_threshold_rt[0].data)
+        self.assertEqual(result[0].coords(), self.multi_threshold_rt[0].coords())
+        assert_array_equal(result[1].data, expected_data)
+        assert_allclose(
+            result[1].coord("probability_bin").points, expected_bin_coord_points
+        )
+        assert_allclose(
+            result[1].coord("probability_bin").bounds, expected_bin_coord_bounds
+        )
+
```

Comment:
test_combine_undersampled_bins_monotonic
test_combine_undersampled_bins_non_monotonic

I think these two tests are currently identical. Have I misread something?

Reply:
Yes, I think you're correct. I've updated the `test_combine_undersampled_bins_non_monotonic`, so that the input data is now non-monotonic, as intended.